prime-angular-frontend/src/app/modules/enrolment/shared/services/enrolment-form-state.service.ts

```diff
@@ -373,28 +373,37 @@ export class EnrolmentFormStateService extends AbstractFormStateService<Enrolmen
     Object.keys(healthAuthoritySites.controls).forEach(healthAuthorityCode => healthAuthoritySites.removeControl(healthAuthorityCode));
 
     oboSites.forEach((s: OboSite) => {
-      const site = this.buildOboSiteForm();
-      site.patchValue(s);
-      oboSitesFormArray.push(site);
-
-      switch (s.careSettingCode) {
-        case CareSettingEnum.PRIVATE_COMMUNITY_HEALTH_PRACTICE: {
-          this.addNonHealthAuthorityOboSite(site, communityHealthSites);
-          break;
-        }
-        case CareSettingEnum.COMMUNITY_PHARMACIST: {
-          this.addNonHealthAuthorityOboSite(site, communityPharmacySites);
-          break;
-        }
-        case CareSettingEnum.DEVICE_PROVIDER: {
-          this.addNonHealthAuthorityOboSite(site, deviceProviderSites);
-          break;
-        }
-        case CareSettingEnum.HEALTH_AUTHORITY: {
-          this.addHealthAuthorityOboSite(site, healthAuthoritySites, s.healthAuthorityCode);
-          break;
+      const careSetting = this.careSettingsForm.get('careSettings');
+
+      if (careSetting && (careSetting.value.length === 0 || careSetting.value.filter((c) => c.careSettingCode === s.careSettingCode).length > 0)) {
+
+        const site = this.buildOboSiteForm();
```

Comment:
Can you make that `siteForm` and add a type to it?  Hmm, I realize how it was before

Reply:
it is fixed

Reply:
Was this updated?  Don't see it

Reply:
![image](https://user-images.githubusercontent.com/108699279/234352105-8371d80c-a8ec-45d3-aa63-2f6529c76b18.png)
The comment is added

Reply:
Ops, the type is added now.